sdk/appcenter-identity/src/main/java/com/microsoft/appcenter/identity/Identity.java

```diff
@@ -429,23 +469,26 @@ private void clearCache() {
 
     private synchronized AppCenterFuture<SignInResult> instanceSignIn() {
         final DefaultAppCenterFuture<SignInResult> future = new DefaultAppCenterFuture<>();
-        if (mPendingSignInFuture != null) {
-            future.complete(new SignInResult(null, new IllegalStateException("signIn already in progress.")));
+        if (isFutureInProgress(mLastSignInFuture)) {
+            future.complete(new SignInResult(null, new IllegalStateException("Sign-in already in progress.")));
             return future;
         }
-        mPendingSignInFuture = future;
+        if (isFutureInProgress(mLastRefreshFuture)) {
+            mLastRefreshFuture.complete(new SignInResult(null, new CancellationException()));
```

Comment:
That should be something like
```suggestion
            mLastRefreshFuture.complete(new SignInResult(null, new IllegalStateException("Silent sign-in already in progress.")));
```
And come to think of it, interactive UI login should take precedence. I think we should discuss that later (just please accept suggestion if it makes sense).